substrate/pkg/utils/discovery/tags.go

```diff
@@ -26,13 +26,13 @@ const (
 	OwnerTagKey = "kit.aws/substrate"
 )
 
-func Tags(substrate *v1alpha1.Substrate, resource string, name *string) []*ec2.TagSpecification {
+func Tags(substrate *v1alpha1.Substrate, resource string, name *string, additionalTags ...*ec2.Tag) []*ec2.TagSpecification {
```

Comment:
I think we should do the append outside, instead of in this method as a variadic.

Reply:
here the `Tags` function returns `[]*ec2.TagSpecification` and we need to append `&ec2.Tag{}` in tagSpecification slice and at the caller side in subnets where we need this functionality it will look like this-

```
tags := []*ec2.Tag{a:b}
if true{
   tags = append(tags, []*ec2.Tag{c:d}
}
t := discovery.Tags(substrate, ec2.ResourceTypeSubnet, name)
t[0].Tags = append(t[0].Tags, tags)
```
which isn't very clean